File: 3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.py
class Solution(object):
    def gcdOfOddEvenSums(self, n):

        sumOdd = n * n
        sumEven = n * (n + 1)

        def gcd(a, b):
            while b:
                a, b = b, a % b
            return a

        return gcd(sumOdd, sumEven)

File: 3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.java
class Solution {
    public int gcdOfOddEvenSums(int n) {
        int even = n * (n + 1);
        int odd = n * n;
        return gcd(odd,even);
    }

    public static int gcd(int a, int b){
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}